torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -87,7 +88,12 @@ def _partition_value(
         if self.reduce_op in (c10d.ReduceOp.MAX, c10d.ReduceOp.MIN):
             return tensor
         elif self.reduce_op == c10d.ReduceOp.SUM:
-            return tensor / mesh.size(mesh_dim=mesh_dim)
```

Comment:
My math was wrong before 😓 .

Consider 2-norm, mesh dim $d$, and vector $v = [v_1, \dots, v_n]$. We want $f: \mathbb R \rightarrow \mathbb R$ such that 
$$\sqrt{v_1^2 + \dots + v_n^2} = \sqrt{df(v_1)^2 + \dots + df(v_n)^2} = \sqrt{d} \sqrt{f(v_1)^2 + \dots + f(v_n)^2}.$$
We see that one possible $f$ is $\frac{1}{\sqrt{d}}$, i.e. divide by $d^{1/2}$. This generalizes for $p$-norm for $p \geq 1$: divide by $d^{1/p}$.

Reply:
could you update the formula to the comment too?